pandas/plotting/_core.py

```diff
@@ -1127,6 +1127,16 @@ def bar(self, x=None, y=None, **kwargs):
         axis of the plot shows the specific categories being compared, and the
         other axis represents a measured value.
         """
+        c = kwargs.pop("c", None)
+        color = kwargs.pop("color", None)
+        if c is not None and color is not None:
+            raise TypeError("Specify exactly one of `c` and `color`")
+        if c is not None or color is not None:
```

Comment:
```suggestion
        elif c is not None or color is not None:
```
is safer for future dev